pkg/protocol/xprotocol/factory.go

```diff
@@ -65,6 +65,20 @@ func GetMatcher(name types.ProtocolName) types.ProtocolMatch {
 	return matcherMap[name]
 }
 
+//GetMatchers return all matchers that was registered
+func GetMatchers() []types.ProtocolMatch {
```

Comment:
Should build `matchers` with `RegisterMatcher`, each request build `matchers` will cost performance.

Reply:
Matchers may be changed by the caller, for this reason, I build it every request. I'll optimize it.

Reply:
> Should build `matchers` with `RegisterMatcher`, each request build `matchers` will cost performance.

Matchers may be changed by the caller, for this reason, I build it every request. I'll optimize it.

Reply:
Why not return `matcherMap` direct?

Reply:
> Why not return `matcherMap` direct?

To protect matcherMap from change by the caller

Reply:
> Why not return `matcherMap` direct?

If it's not necessary.I'll return matcherMap direct.